cypress/e2e/add_assurance_visit.cy.js

```diff
@@ -28,12 +28,20 @@ describe("Add Assurance Visit", () => {
 
     describe("Error submitting assurance visit form", () => {
         it("shows error message when submitting invalid data", () => {
-        cy.setCookie("fail-route", "addAssuranceVisit");
-        cy.get("#add-assurance-visit-form").submit();
-        cy.get(".govuk-error-summary__body").should(
-            "contain",
-            "Enter a requested date"
+            cy.setCookie("fail-route", "addAssuranceVisit");
+            cy.get("#add-assurance-visit-form").submit();
+            cy.get(".govuk-error-summary__body").should(
+                "contain",
+                "Select an assurance type"
+            ).should(
+                "contain",
+                "Enter a requested date"
             );
+            cy.get("#f-requested-date").should("have.class", "govuk-input--error")
+            cy.get("#add-assurance-visit-form > :nth-child(2)").should("have.class", "govuk-form-group--error")
```

Comment:
`.should("have.class"` is redundant as you can just include the class you are trying to assert on in the `get`.

This is also a very brittle solution to checking for form errors. If the order of the data changes or the form fields move around, this test would break inadvertently (or not break when it should).